src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -22,135 +104,791 @@ const ActorInit En_Pamera_InitVars = {
+                func_800BDC5C(&this->skelAnime, sAnimations, 1);
+                this->actor.shape.rot.y = this->actor.home.rot.y;
+                this->actor.world.rot.y = this->actor.home.rot.y;
+                func_80BD9338(this, globalCtx);
+                func_80BD8908(this);
+            }
+        }
+    }
+    if ((this->actor.draw != NULL) && (this->actionFunc != func_80BD8964)) {
+        if ((this->actor.child != NULL) && (this->actor.child->id == ACTOR_EN_DOOR) && (this->actor.child->world.rot.y == 0)) {
+            this->actor.draw = NULL;
+        }
+    }
+    if (func_80BD9234(this, globalCtx)) {
+        // this is 1760 frames, which is ~90sec
+        this->hideInisdeTimer = 1760;
+    }
+}
+
+void func_80BD8908(EnPamera* this) {
+    this->actor.draw = EnPamera_Draw;
+    this->actor.flags |= 1;
+    func_800BDC5C(&this->skelAnime, sAnimations, 1);
+    this->actionFunc = func_80BD8964;
+}
+
+void func_80BD8964(EnPamera* this, GlobalContext* globalCtx) {
+    Vec3f vec;
+
+    vec.x = this->pathPoints->x;
+    vec.y = this->pathPoints->y;
+    vec.z = this->pathPoints->z;
+
+    if (Math_Vec3f_StepTo(&this->actor.world.pos, &vec, 1.0f) < 5.0f) {
+        this->actor.speedXZ = 1.5f;
+        func_800BDC5C(&this->skelAnime, sAnimations, 1);
+        gSaveContext.weekEventReg[59] |= 1;
+        func_80BD8B50(this);
+    }
+}
+
+void func_80BD8A38(EnPamera* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, 1);
+    this->actionFunc = func_80BD8A7C;
+}
+
+void func_80BD8A7C(EnPamera* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.home.pos), 0xA,
+                       0x3000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if (Math_Vec3f_StepTo(&this->actor.world.pos, &this->actor.home.pos, 1.5f) < 10.0f) {
+        gSaveContext.weekEventReg[59] &= (u8)~1;
+        if (!(gSaveContext.weekEventReg[61] & 4)) {
+            func_80BD92D0(this, globalCtx);
+            gSaveContext.weekEventReg[61] |= 4;
+        }
+        func_80BD8700(this);
+    }
+}
+
+void func_80BD8B50(EnPamera* this) {
+    this->actionFunc = func_80BD8B70;
+    this->actor.gravity = -2.0f;
+}
+
+void func_80BD8B70(EnPamera* this, GlobalContext* globalCtx) {
+    Vec3f vec;
+    s16 sp32;
+
+    vec.x = this->pathPoints[this->pathIndex].x;
+    vec.y = this->pathPoints[this->pathIndex].y;
+    vec.z = this->pathPoints[this->pathIndex].z;
+    sp32 = Math_Vec3f_Yaw(&this->actor.world.pos, &vec);
+    if (Math_Vec3f_StepToXZ(&this->actor.world.pos, &vec, this->actor.speedXZ) > 10.0f) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, sp32, 0xA, 0x3000, 0x100);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    } else {
+        if (this->pathIndex < (this->pathPointsCount - 1)) {
+            this->pathIndex++;
+        } else {
+            func_80BD8CCC(this);
+        }
+    }
```

Comment:
You could mix those into one `else if` if you think it makes more sense. Up to you.
```suggestion
    } else if (this->pathIndex < (this->pathPointsCount - 1)) {
        this->pathIndex++;
    } else {
        func_80BD8CCC(this);
    }
```